DigitalLearningSolutions.Web/Views/Login/Index.cshtml

```diff
@@ -1,12 +1,82 @@
-﻿@{
+﻿@using DigitalLearningSolutions.Web.ViewModels.Login
+@model LoginViewModel 
+@{
   ViewData["Title"] = "Login";
+  var errorHasOccurred = !ViewData.ModelState.IsValid;
+  var usernameError = ViewData.ModelState["Username"]?.Errors?.Count > 0;
+  var passwordError = ViewData.ModelState["Password"]?.Errors?.Count > 0;
+  var passwordInputErrorClass = passwordError ? "nhsuk-input--error" : "";
+  var passwordFormErrorClass = passwordError ? "nhsuk-form-group--error" : "";
+  var usernameInputErrorClass = usernameError ? "nhsuk-input--error" : "";
+  var usernameFormGroupError = usernameError ? "nhsuk-form-group--error" : "";
 }
 
 <div class="nhsuk-grid-row">
   <div class="nhsuk-grid-column-full">
-    <h1 id="page-heading">Login</h1>
-    <a class="nhsuk-button" asp-controller="Login" asp-action="SignIn">
-      Sign in
+    <form method="post" novalidate asp-action="Login">
+
+      @if (errorHasOccurred) {
+        <partial name="_ErrorSummary"/>
+      }
+
+      <h1 id="page-heading">Login</h1>
+      <p>Enter your details to log into your account.</p>
+
+      <div class="nhsuk-form-group @usernameFormGroupError">
+        <label class="nhsuk-label" for="Username">
+          Email or user ID
+        </label>
+        @if (usernameError) {
+          <span class="nhsuk-error-message" id="username-error">
+            @foreach (var error in ViewData.ModelState["Username"].Errors) {
+              <span class="nhsuk-u-visually-hidden">Error:</span> @error.ErrorMessage
+            }
+          </span>
+        }
+        <input class="nhsuk-input nhsuk-u-width-one-half nhsuk-u-margin-bottom-4 @usernameInputErrorClass" type="text" id="Username" spellcheck="false" autocomplete="email" aria-describedby="username-error" asp-for="Username" />
```

Comment:
I think you forgot the email autocomplete here

Reply:
The ticket mentions that the username input is "Email/UserId text input" - would the chance of input not being an email address (e.g. UserId) mean that the autocomplete="email" should be removed?